web/packages/design/src/SlideTabs/SlideTabs.tsx

```diff
@@ -112,77 +164,189 @@ type props = {
    * If true, disables pointer events.
    */
   disabled?: boolean;
+  /**
+   * If true, the control doesn't take as much horizontal space as possible,
+   * but instead wraps its contents.
+   */
+  fitContent?: boolean;
 };
 
-export type TabComponent = {
-  name: string;
-  component: React.ReactNode;
+/**
+ * Definition of a tab. If it's a string, it denotes a title displayed on the
+ * tab. It's recommended to use a full object with tab panel ID for better
+ * accessibility.
+ *
+ * TODO(bl-nero): remove the string option once Enterprise is migrated to
+ * simplify it a bit.
+ */
+type TabSpec = string | FullTabSpec;
+
+type FullTabSpec = TabContentSpec & {
+  /** Iteration key for the tab. */
+  key: React.Key;
+  /**
+   * ID of the controlled element for accessibility, perhaps autogenerated
+   * with an `useId()` hook. The indicated element should have a `role`
+   * attribute set to "tabpanel".
+   */
+  controls?: string;
 };
 
-const Wrapper = styled.div`
+/**
+ * Tab content. Either an icon with a mandatory accessible label, or a
+ * decorative icon with accompanying text.
+ */
+type TabContentSpec =
+  | {
+      /** Title displayed on the tab. */
+      title: string;
+      ariaLabel?: never;
```

Comment:
I'm wondering if we should allow `ariaLabel` when providing the title.  However, I'm not sure how practical or useful this would be in real-world scenarios (probably no one would use it anyway).

Reply:
@gzdunek IMO, `ariaLabel` doesn't make any sense when the title is here.